src/backend/distributed/commands/table.c

```diff
@@ -1762,14 +1788,22 @@ PostprocessAlterTableStmt(AlterTableStmt *alterTableStatement)
 						ClusterHasKnownMetadataWorkers())
 					{
 						MarkSequenceDistributedAndPropagateDependencies(seqOid);
+
+						AppendAlterTableWithNextvalDefaultStmt(
+							&alterTableDefaultNextvalStmt,
+							seqOid, relationId,
+							command->name, NULL);
 					}
 				}
 			}
 		}
 	}
 
 	/* for the new sequences coming with this ALTER TABLE statement */
-	if (ShouldSyncTableMetadata(relationId) && ClusterHasKnownMetadataWorkers())
+	bool newSequencesMetadataSync = alterTableDefaultNextvalStmt.len > 0 &&
+									ShouldSyncTableMetadata(relationId) &&
+									ClusterHasKnownMetadataWorkers();
```

Comment:
I was thinking we would set newSequencesMetadataSync to true in the places where we generate alterTableDefaultNextvalStmt, which are already conditional on `ShouldSyncTableMetadata(relationId) &&	ClusterHasKnownMetadataWorkers()`

Reply:
Right, much better to do it there